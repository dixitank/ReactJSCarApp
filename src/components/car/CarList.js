import React, { PropTypes } from 'react';
import { BootstrapTable, TableHeaderColumn } from 'react-bootstrap-table';



const getCaret = direction => {
    if (direction === 'asc') {
        return (
            <span> <i className="fa fa-sort-asc" aria-hidden="true"/></span>
        );
    }

    if (direction === 'desc') {
        return (
            <span> <i className="fa fa-sort-desc" aria-hidden="true"/></span>
        );
    }

    return (
        <span> <i className="fa fa-sort" aria-hidden="true" /></span>
    );
};



const titleFormatter = (cell, row) => {
  return `<a href=${row.watchHref} target="_blank">${cell}</a>`;
};



class CarList extends React.Component {

    constructor(props) {
        super(props);

        this.options = {
            sortIndicator: true,
            noDataText: 'No data'
        };

        this.selectRowProp = {
            mode: 'radio',
            bgColor: '#c1f291',
            clickToSelect: true, 
            hideSelectColumn: true            
        };
    }



    render() {
        let years = [...new Set(this.props.cars.map(car=> car.yearOfRelease))];
        return (
            <BootstrapTable data={this.props.cars}  selectRow={this.selectRowProp}  options={this.options} bordered={false} striped hover condensed>
                <TableHeaderColumn dataField="id" isKey hidden>Id</TableHeaderColumn>
                
                <TableHeaderColumn 
                    dataField="model"
                    dataSort={true}
                    caretRender={getCaret}
                    columnTitle
                >
                    Model
                </TableHeaderColumn>

                <TableHeaderColumn 
                    dataField="brand"
                    dataSort={true}
                    caretRender={getCaret}
                    filter={{type: 'TextFilter', condition: 'eq', delay: 0 }}
                    columnTitle
                >
                    Brand
                </TableHeaderColumn>

                <TableHeaderColumn 
                    dataField="version"
                    dataSort={true}
                    caretRender={getCaret}
                    columnTitle
                >
                    Version
                </TableHeaderColumn>  

                <TableHeaderColumn 
                    dataField="yearOfRelease"
                    dataSort={true}
                    caretRender={getCaret}
                    columnTitle
                    filter={{type: 'NumberFilter',options: years}}
                >
                    Year Of Release
                </TableHeaderColumn>  
                
                <TableHeaderColumn 
                    dataField="price"
                    dataSort={true}
                    caretRender={getCaret}
                    columnTitle
                >
                    price
                </TableHeaderColumn>    
                
                <TableHeaderColumn 
                    dataField="fuelConsumption"
                    dataSort={true}
                    caretRender={getCaret}
                    columnTitle
                >
                    Fuel Consumption
                </TableHeaderColumn> 
                
                <TableHeaderColumn 
                    dataField="annualMaintenanceCost"
                    dataSort={true}
                    caretRender={getCaret}
                    columnTitle
                >
                    Annual Maintainence Cost
                </TableHeaderColumn>                               
            </BootstrapTable>
        );
    }

}



CarList.propTypes = {
    cars: PropTypes.array.isRequired
};



export default CarList;
