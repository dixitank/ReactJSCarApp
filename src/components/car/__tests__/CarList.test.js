import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import CarList from '../CarList';


describe('CarList.test.js', () => {

    it('renders without crashing', () => {
        const props = {
            cars: [
                { brand: 'Ford', Model: 'Ford' },
                { brand: 'Fusion', Model: 'Fusion' },              
            ],
            handleRowSelect: jest.fn()            
        };
        
        const wrapper = shallow(<CarList {...props}/>);

        expect(wrapper).toHaveLength(1);
        expect(wrapper.find('BootstrapTable')).toHaveLength(1);

        const tree = toJson(wrapper);
        expect(tree).toMatchSnapshot();
    });
});
