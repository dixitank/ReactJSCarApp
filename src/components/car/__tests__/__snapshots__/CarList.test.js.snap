exports[`CarList.test.js renders without crashing 1`] = `
<BootstrapTable
  bodyContainerClass={null}
  bordered={false}
  cellEdit={
    Object {
      "afterSaveCell": undefined,
      "beforeSaveCell": undefined,
      "blurToSave": false,
      "mode": "none",
      "nonEditableRows": undefined,
    }
  }
  columnFilter={false}
  condensed={true}
  containerClass={null}
  csvFileName="spreadsheet.csv"
  data={
    Array [
      Object {
        "Model": "Ford",
        "brand": "Ford",
      },
      Object {
        "Model": "Fusion",
        "brand": "Fusion",
      },
    ]
  }
  deleteRow={false}
  expandColumnOptions={
    Object {
      "expandColumnBeforeSelectColumn": true,
      "expandColumnComponent": undefined,
      "expandColumnVisible": false,
    }
  }
  exportCSV={false}
  fetchInfo={
    Object {
      "dataTotalSize": 0,
    }
  }
  headerContainerClass={null}
  height="100%"
  hover={true}
  ignoreSinglePage={false}
  insertRow={false}
  keyBoardNav={false}
  multiColumnSearch={false}
  multiColumnSort={1}
  options={
    Object {
      "noDataText": "No data",
      "sortIndicator": true,
    }
  }
  pagination={false}
  printable={false}
  replace={false}
  search={false}
  selectRow={
    Object {
      "bgColor": "#c1f291",
      "clickToSelect": true,
      "hideSelectColumn": true,
      "mode": "radio",
    }
  }
  striped={true}
  tableBodyClass={null}
  tableContainerClass={null}
  tableHeaderClass={null}
  trClassName=""
  withoutTabIndex={false}>
  <TableHeaderColumn
    className=""
    columnClassName=""
    columnTitle={false}
    dataAlign="left"
    dataField="id"
    dataSort={false}
    defaultASC={false}
    editColumnClassName=""
    editable={true}
    expandable={true}
    filterFormatted={false}
    headerTitle={true}
    hidden={true}
    hiddenOnInsert={false}
    invalidEditColumnClassName=""
    isKey={true}
    keyValidator={false}
    searchable={true}
    sortIndicator={true}
    width={null}>
    Id
  </TableHeaderColumn>
  <TableHeaderColumn
    caretRender={[Function]}
    className=""
    columnClassName=""
    columnTitle={true}
    dataAlign="left"
    dataField="model"
    dataSort={true}
    defaultASC={false}
    editColumnClassName=""
    editable={true}
    expandable={true}
    filterFormatted={false}
    headerTitle={true}
    hidden={false}
    hiddenOnInsert={false}
    invalidEditColumnClassName=""
    isKey={false}
    keyValidator={false}
    searchable={true}
    sortIndicator={true}
    width={null}>
    Model
  </TableHeaderColumn>
  <TableHeaderColumn
    caretRender={[Function]}
    className=""
    columnClassName=""
    columnTitle={true}
    dataAlign="left"
    dataField="brand"
    dataSort={true}
    defaultASC={false}
    editColumnClassName=""
    editable={true}
    expandable={true}
    filter={
      Object {
        "condition": "eq",
        "delay": 0,
        "type": "TextFilter",
      }
    }
    filterFormatted={false}
    headerTitle={true}
    hidden={false}
    hiddenOnInsert={false}
    invalidEditColumnClassName=""
    isKey={false}
    keyValidator={false}
    searchable={true}
    sortIndicator={true}
    width={null}>
    Brand
  </TableHeaderColumn>
  <TableHeaderColumn
    caretRender={[Function]}
    className=""
    columnClassName=""
    columnTitle={true}
    dataAlign="left"
    dataField="version"
    dataSort={true}
    defaultASC={false}
    editColumnClassName=""
    editable={true}
    expandable={true}
    filterFormatted={false}
    headerTitle={true}
    hidden={false}
    hiddenOnInsert={false}
    invalidEditColumnClassName=""
    isKey={false}
    keyValidator={false}
    searchable={true}
    sortIndicator={true}
    width={null}>
    Version
  </TableHeaderColumn>
  <TableHeaderColumn
    caretRender={[Function]}
    className=""
    columnClassName=""
    columnTitle={true}
    dataAlign="left"
    dataField="yearOfRelease"
    dataSort={true}
    defaultASC={false}
    editColumnClassName=""
    editable={true}
    expandable={true}
    filter={
      Object {
        "options": Array [
          undefined,
        ],
        "type": "NumberFilter",
      }
    }
    filterFormatted={false}
    headerTitle={true}
    hidden={false}
    hiddenOnInsert={false}
    invalidEditColumnClassName=""
    isKey={false}
    keyValidator={false}
    searchable={true}
    sortIndicator={true}
    width={null}>
    Year Of Release
  </TableHeaderColumn>
  <TableHeaderColumn
    caretRender={[Function]}
    className=""
    columnClassName=""
    columnTitle={true}
    dataAlign="left"
    dataField="price"
    dataSort={true}
    defaultASC={false}
    editColumnClassName=""
    editable={true}
    expandable={true}
    filterFormatted={false}
    headerTitle={true}
    hidden={false}
    hiddenOnInsert={false}
    invalidEditColumnClassName=""
    isKey={false}
    keyValidator={false}
    searchable={true}
    sortIndicator={true}
    width={null}>
    price
  </TableHeaderColumn>
  <TableHeaderColumn
    caretRender={[Function]}
    className=""
    columnClassName=""
    columnTitle={true}
    dataAlign="left"
    dataField="fuelConsumption"
    dataSort={true}
    defaultASC={false}
    editColumnClassName=""
    editable={true}
    expandable={true}
    filterFormatted={false}
    headerTitle={true}
    hidden={false}
    hiddenOnInsert={false}
    invalidEditColumnClassName=""
    isKey={false}
    keyValidator={false}
    searchable={true}
    sortIndicator={true}
    width={null}>
    Fuel Consumption
  </TableHeaderColumn>
  <TableHeaderColumn
    caretRender={[Function]}
    className=""
    columnClassName=""
    columnTitle={true}
    dataAlign="left"
    dataField="annualMaintenanceCost"
    dataSort={true}
    defaultASC={false}
    editColumnClassName=""
    editable={true}
    expandable={true}
    filterFormatted={false}
    headerTitle={true}
    hidden={false}
    hiddenOnInsert={false}
    invalidEditColumnClassName=""
    isKey={false}
    keyValidator={false}
    searchable={true}
    sortIndicator={true}
    width={null}>
    Annual Maintainence Cost
  </TableHeaderColumn>
</BootstrapTable>
`;
